packages/shared.net/src/ClientContext.ts

```diff
@@ -14,21 +14,45 @@
  * limitations under the License.
  */
 
-export interface ClientContext<O extends { metadata: { userAgent: string } }> {
+/** @deprecated */
+export interface ClientContext<O extends { metadata: { userAgent: string } }>
+  extends SharedClientContext
+{
   /** @deprecated */
   ontology: O;
 
   /**
    * The base origin to use for requests (e.g. `https://api.example.com`)
+   * @deprecated use `baseUrl` instead
    */
   stack: string;
+}
+
+/**
+ * Strict representation of ClientContext. This is the type that should be used.
+ *
+ * Upon next major version bump, this should be merged with `ClientContext`.
+ */
+export interface SharedClientContext {
+  /**
+   * The base origin to use for requests (e.g. `https://api.example.com`)
+   */
+  baseUrl: string;
 
   /**
    * The fetch function to use for all requests.
-   *
-   * TODO: Document what is needed to get retry logic
    */
   fetch: typeof globalThis.fetch;
 
-  tokenProvider: () => Promise<string> | string;
+  tokenProvider: () => Promise<string>;
```

Comment:
This switches to `Promise<string>` as its both back compat with the old one (you had to `await` either way) and it makes it assignable into `ConjureContext`.